src/main/java/uk/gov/register/core/RegisterMemoizationStoreFactory.java

```diff
@@ -0,0 +1,22 @@
+package uk.gov.register.core;
+
+import uk.gov.verifiablelog.store.memoization.InMemory;
+import uk.gov.verifiablelog.store.memoization.InMemoryPowOfTwoNoLeaves;
+import uk.gov.verifiablelog.store.memoization.MemoizationStore;
+
+public class RegisterMemoizationStoreFactory {
+    private String memoizationStoreTypeName;
+
+    public <T extends MemoizationStore> RegisterMemoizationStoreFactory(Class<T> memoizationStoreClass) {
+        this.memoizationStoreTypeName = memoizationStoreClass.getSimpleName();
```

Comment:
I think it's good to think about how we might configure the specific MemoizationStore that we use, rather than hardcoding, but I don't think we should do that as part of fixing a bug.

Let's have a conversation about this.